### DISCUSSION TITLE
How to type errors object in typescript?

### DISCUSSION THREAD

**Discussion by mimoid-prog** [react-hook-form/react-hook-form#7179] [Q&A]
Hi, 
I'm trying to do custom Input component which accepts props from the parent like `name`, `register`, `errors` etc.
My problem is that I cannot type `errors` object correctly and I get an error:
`Type 'Path<T>' cannot be used to index type 'DeepMap<DeepPartial<UnionLike<T>>, FieldError>'.`

In previous versions I made it work like this `name: FieldPath<T>; errors: FieldErrors<T>`, but in the version `7.15.4` it doesn't work anymore.

Link to full code: https://codesandbox.io/s/react-hook-form-get-started-ts-forked-jwfmg?file=/src/Input.tsx:0-497

**Comment by Joe312341** [comment]
Facing the same issue

**Comment by bluebill1049** [comment]
looks like this issue is due to the union like object support, let me see if we can work around this type problem.

**Reply by mimoid-prog** [reply]
Have you found any solution?

**Reply by Joe312341** [reply]
Looks like they are working on a PR: https://github.com/react-hook-form/react-hook-form/pull/7187
